### DISCUSSION TITLE
Outdated Settings Documentation / Portable settings documentation

### DISCUSSION THREAD

**Discussion by docfresh** [dotnet/winforms#6178] [Q&A]
The Application Settings help file [here ](https://docs.microsoft.com/en-us/dotnet/desktop/winforms/advanced/application-settings-for-windows-forms?view=netframeworkdesktop-4.8&WT.mc_id=DT-MVP-5003235)is outdated. It refers to .NET 4 but not 5 or 6.  I can't find an updated version.  Some of the relevant info is [here](https://stackoverflow.com/questions/65518159/where-are-the-settings-saved-in-a-net-5-winforms-app).

Also, it would be great if an updated doc could give recommended methods for making a settings file portable; for example, read/write from the root directory of a USB drive, instead of in the Windows user's appdata folder. 

Hacks for this could probably be figured out, but enough people may be interested in portable applications that is is worth including in the docs.

Thanks

**Comment by RussKie** [comment]
There hasn't been any significant changes between .NET Framework and .NET as far as settings or resources are concerned. So pretty much all the docs are still valid and relevant.
The referenced SO issue is a [known CPS bug](https://github.com/dotnet/project-system/issues/7772).

> Also, it would be great if an updated doc could give recommended methods for making a settings file portable; for example, read/write from the root directory of a USB drive, instead of in the Windows user's appdata folder.

The _Settings_ have two distinct modes - application-scope (i.e., system-wide, stored in *.exe.config) and user-specific (stored under the user profile). In portable scenarios you can use the application-scope and persist into the *.exe.config as it would likely be user-writable.

**Reply by adegeo** [reply]
@docfresh The difference between user/application settings isn't called out very well in the .NET Framework docs. I'll put a note in the [migration issue on github](https://github.com/dotnet/docs-desktop/issues/72) for this section. As we get to migrating these from .NET Framework to .NET Core, we're improving the content, removing dead articles, and retelling the story.
